src/plugins/ResizeReorder/ResizerKnob.js

```diff
@@ -0,0 +1,170 @@
+        instance={this.state.instance}
+        xCoordinate={this.state.resizerLineXCoordinate}
+      />;
+
+    return (
+      <div
+        className={cx('fixedDataTableCellLayout/columnResizerContainer')}
+        ref={(element) => this.curRef = element}
+        style={this.props.columnResizerStyle}
+        onMouseDown={this.onMouseDown}
+        onTouchStart={this.props.touchEnabled ? this.onMouseDown : null}
+        onTouchEnd={this.props.touchEnabled ? this.suppressEvent : null}
+        onTouchMove={this.props.touchEnabled ? this.suppressEvent : null}>
+        {dragKnob}
+        <div
+          className={joinClasses(
+            cx('fixedDataTableCellLayout/columnResizerKnob'),
+            cx('public/fixedDataTableCell/columnResizerKnob'),
+          )}
+          style={this.props.columnResizerStyle}
+        />
+      </div>
+    );
+  }
+
+  /**
+   * Registers event listeners for mouse tracking
+   */
+  trackMouse = () => {
+    this._mouseMoveTracker = new DOMMouseMoveTracker(
+      this.onMouseMove,
+      this.onMouseUp,
+      document.body,
+      this.props.touchEnabled
+    );
+  };
+
+  /**
+   * @param {MouseEvent} ev Mouse down event
+   */
+  onMouseDown = (ev) => {
+    const initialMouseCoordinates = FixedDataTableEventHelper.getCoordinatesFromEvent(ev);
+    this._mouseMoveTracker.captureMouseMoves(ev);
+    this.setState({
+      initialMouseCoordinates,
+      isColumnResizing: true,
+      totalDisplacement: 0,
+      resizerLineXCoordinate: initialMouseCoordinates.x
+    });
+  };
+
+  onMouseUp = () => {
+    const { minWidth, maxWidth } = this.getMinMaxWidth(this.props.minWidth, this.props.maxWidth);
+    const newWidth = clamp(this.props.width + this.state.totalDisplacement, minWidth, maxWidth);
+    this.props.onColumnResizeEnd(newWidth, this.props.columnKey);
+    this._mouseMoveTracker.releaseMouseMoves();
+    this.resetColumnResizing();
+  };
+
+  /**
+   *
+   * @param {number} displacementX Displacement of mouse along x-direction
+   */
+  onMouseMove = (displacementX) => {
+    const { initialMouseCoordinates, totalDisplacement: previousTotalDisplacement } = this.state;
+    // displacementX is negative if movement is in left direction
+    const newTotalDisplacement = previousTotalDisplacement + displacementX;
+    let resizerLineXCoordinate = initialMouseCoordinates.x + newTotalDisplacement;
+    const { minWidth, maxWidth } = this.getMinMaxWidth(this.props.minWidth, this.props.maxWidth);
+    // Limit the resizer line to not move ahead or back of maxWidth and minWidth respectively
+    if (this.props.width + newTotalDisplacement < minWidth || this.props.width + newTotalDisplacement > maxWidth) {
+      // If new position is going out of bounds, instead of updating,  use the previous value
+      resizerLineXCoordinate = this.state.resizerLineXCoordinate;
+    }
+    this.setState({
+      mouseXCoordinate: resizerLineXCoordinate,
+      isColumnResizing: true,
```

Comment:
do we need to set `isColumnResizing` here? Should we instead just rely on `mouseDown` for that?

Reply:
Done, removed it